### PR TITLE
Component object backend props

### PR DESCRIPTION
**Pull request by weaverryan**
Accomplishes `3) BACKEND: pass data (writable props) & props separately as live controller values` from https://github.com/symfony/ux/pull/500

tl;dr

We now pass `props` (readonly `LiveProp`) and `data` (writable `LiveProp`) separately to the frontend. But then when we re-render, we still send back just one big array of BOTH mixed together.

Why do we need to differentiate between `props` and `data` on the frontend? It's for child components. When a parent re-renders, we will (it's another TODO on the PR) render the parent but with an EMPTY child element, except that we will include the `props` on that child element - e.g. this will come back from the Ajax request:

```
<div data-controller="live" ... data & props - all the normal attributes>
    I am a parent component

    <!-- here is the child - it's empty
    <div data-controller="live" data-live-props-value="{...}"></div>
</div>
```

This will allow us to grab the new props (which may have changed as a result of WHAT the parent component is passing into the child), override the props on the child, and then trigger the child to re-render with all the new stuff. What's important here is that, by differentiating props vs data, we can KEEP any modified "data" that the child has: only overriding the *props* that changed as a result of the parent component re-rendering.

### CODE REVIEW

File: src/LiveComponent/src/LiveComponentHydrator.php [hunk 1]
```diff
@@ -209,23 +198,20 @@ public function hydrate(object $component, array $data, string $componentName):
         return new MountedComponent($componentName, $component, $attributes);
     }
 
-    private function computeChecksum(array $data, array $readonlyProperties): string
+    private function computeChecksum(array $props): string
```
Comment (kbond) [inline src/LiveComponent/src/LiveComponentHydrator.php]: Sort of thinking checksum calculation/verification should on DehydratedComponent

**Review by kbond — COMMENTED** [review]
Can more logic be moved to the new DehydratedComponent class?

**Review by kbond — APPROVED** [review]
LGTM!
